### DISCUSSION TITLE
Feature Request: Support create directories at write time

### DISCUSSION THREAD

**Discussion by dchimeno** [duckdb/duckdb#10665] [Ideas]
It would be useful to treat external tables and local filesystem tables the same way:

```
COPY (select 42 AS number) TO 's3://bucket/data/test/tes.parquet' (FORMAT 'parquet');
-- it works
```

```
COPY (select 42 AS number) TO 'data/test/tes.parquet' (FORMAT 'parquet') ;
--Error: IO Error: Cannot open file "data/test/tes.parquet": No such file or directory
```

After:
```bash
mkdir -p data/test
```
the query above works.

Just wondering if it's possible to include this,  it seems better DX to me.

**Comment by Tishj** [comment]
Maybe as an additional key-value pair to the COPY function, like FORMAT in your example ?

**Reply by dchimeno** [reply]
I would expect it to be the default behavior and work across all writing functions.
But yeah, I parameter would be better than nothing ;)

**Reply by claudino-eb** [reply]
I agree. This should be the default behavior.

**Comment by orzom411** [comment]
In the CLI I often use .shell, e.g.:

`.shell "if not exist .\20240409 md .\20240409"`

This works okay in windows, Linux I believe mkdir just does the deed if needed:

`.shell "mkdir -p ./20240409"`

Not sure if DuckDB should default to creating sub-folders.  With S3 it is known to be a key value pair, e.g. the key points to a blob so it isn't a "folder" at all.  For file systems, on the other hand, position is relative.  Maybe a happy middle is fully qualified assumption is to create the path, otherwise explicitly stating to create the folder might be a predictable solution?  Just a thought.

**Comment by djouallah** [comment]
it is just very annoying :( specially with s3, as it will introduce an extra dependency just for creating a folder
